src/utils/daos_hdlr.c

```diff
@@ -2649,15 +2653,15 @@ dm_connect(struct cmd_args_s *ap,
 					    dst_cont_info, NULL);
 		else
 			rc = -DER_NONEXIST;
+
 		if (rc == -DER_NONEXIST) {
 			uuid_t cuuid;
 
 			if (ca->cont_layout == DAOS_PROP_CO_LAYOUT_POSIX) {
+				/** XXX: should copy dfs_attr & properties from source container */
 				if (dst_cont_passed) {
-					rc = uuid_parse(ca->dst_cont, cuuid);
```

Comment:
@johannlombardi why is this restricting the user from passing in a container UUID? I am curious why this is not optional to copy by passing in UUID or label?

This is going to break all of the datamover functional tests that pass in a container UUID for the destination:

```bash
daos container clone --dst=daos://A7348772-DAE1-4227-95DB-9E8D18A592C5/f9422eeb-4d68-4a9c-a2a5-ac4b378b0f7c/ --src=daos://A7348772-DAE1-4227-95DB-9E8D18A592C5/3A29D6C7-9991-4372-8320-F03995A8C0D6/' failed.
stdout: b''
stderr: b'ERROR: : failed to create destination container: Invalid parameters (-1003)\nERROR: : failed to close source container: Invalid handle (-1002)\nERROR: : failed to disconnect source pool: Invalid handle (-1002)\nERROR: : failed to close destination container: Invalid handle (-1002)\nERROR: : failed to disconnect destination pool: Invalid handle (-1002)\nERROR: : Failed to disconnect: Invalid handle (-1002)\nERROR: daos: failed to clone container daos://A7348772-DAE1-4227-95DB-9E8D18A592C5/3A29D6C7-9991-4372-8320-F03995A8C0D6/: DER_INVAL(-1003): Invalid parameters\n'
```
 Also, if a user does not pass in a a container for the destination (just a pool) then a container is always created by generating a new UUID. 

If you add:
Test-tag: pr fs_copy cont_clone

It will show all of the datamover functional tests that break because of not being able to pass in a destination container UUID.

Reply:
ok, i will look at those tests. The reason for this change is that, now that users can use labels, UUIDs are always generated internally. Allowing users to choose both the label and UUID is troublesome.

Reply:
@johannlombardi I ran most of the datamover functional tests on this PR locally (except for ones with "large" in the name), and I don't think those matter since those are typically multi-node tests only for mpifileutils/dcp. I found failing tests in dm_dst_create.py and dm_obj_small.py. I pasted the output from dm_obj_small.py in the above comment, but I would also add fs_copy and cont_clone to test-tag in the commit to run them in CI as well.